Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.cpp

```diff
@@ -267,6 +267,30 @@ namespace ROS2::Utils
         return availableAssets;
     }
 
+    AZ::Outcome<AzToolsFramework::AssetSystem::JobInfoContainer> WaitForAPAndEscalate(
+        const AZStd::string& relativePath, const AZStd::chrono::duration<int, AZStd::milli>& timeout)
+    {
+        const auto loopTime = AZStd::chrono::system_clock::now();
+        using namespace AzToolsFramework;
+        using namespace AzToolsFramework::AssetSystem;
+        AZ::Outcome<AssetSystem::JobInfoContainer> result = AZ::Failure();
+
+        while (loopTime - AZStd::chrono::system_clock::now() < timeout)
+        {
+            AssetSystemJobRequestBus::BroadcastResult(result, &AssetSystemJobRequestBus::Events::GetAssetJobsInfo, relativePath, true);
+            if (result.IsSuccess())
+            {
+                return result;
+            }
+        }
+        AZ_Warning(
+            "WaitForAPAndEscalate",
+            false,
+            "Cannot find job for %s, import may be incomplete or AssetProcessor is not running.",
+            relativePath.c_str());
+        return result;
+    }
```

Comment:
Rather than calling this request bus in a tight loop, CompileAssetSync will attempt to immediately compile the asset and block until there is a result.
```
                    AzFramework::AssetSystem::AssetStatus status = AzFramework::AssetSystem::AssetStatus_Unknown;
                    AzFramework::AssetSystemRequestBus::BroadcastResult(
                        status, &AzFramework::AssetSystemRequestBus::Events::CompileAssetSync_FlushIO, relativePath);
```

Reply:
Maybe we should consider `GetAssetStatusById_FlushIO` here?